### PR TITLE
Add check to override submissions visibility for private phase

### PR DESCRIPTION
**Pull request by Ram81**
### Description

This PR -

- [x] Adds check to override submission visibility passed from cli when phase is private

### CODE REVIEW

File: apps/jobs/views.py [hunk 1]
```diff
@@ -336,6 +336,10 @@ def challenge_submission(request, challenge_id, challenge_phase_id):
                     if submission_serializer.is_valid():
                         submission_serializer.save()
 
+        # Override submission visibility if leaderboard_public = False for a challenge phase
+        if not challenge_phase.leaderboard_public:
+            request.data["is_public"] = False
```
Comment (RishabhJain2018) [inline apps/jobs/views.py:341]: Let's not hardcode this but pick from the `challenge_phase` object.

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]

**Comment by RishabhJain2018** [comment]
@Ram81 Can you please update this PR?

**Comment by Ram81** [comment]
@RishabhJain2018 done

File: apps/jobs/views.py [hunk 2]
```diff
@@ -336,6 +336,10 @@ def challenge_submission(request, challenge_id, challenge_phase_id):
                     if submission_serializer.is_valid():
                         submission_serializer.save()
 
+        # Override submission visibility if leaderboard_public = False for a challenge phase
+        if not challenge_phase.leaderboard_public:
+            request.data["is_public"] = challenge_phase.leaderboard_public
```
Comment (RishabhJain2018) [inline apps/jobs/views.py:341]: It should be `challenge_phase.leaderboard_public` -> `challenge_phase.is_submission_public`, no?

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]

File: apps/jobs/views.py (hunk 2 above)
Comment (Ram81) [reply apps/jobs/views.py:341]: is it possible for hosts to set `leaderboard_public=False` and `is_submission_public=True` ? both will be same right?

Comment (RishabhJain2018) [reply apps/jobs/views.py:341]: Yes, it is possible for them to set these things and they both are different fields.

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**